client/signup/steps/site-or-domain/index.jsx

```diff
@@ -120,8 +128,11 @@ class SiteOrDomain extends Component {
 	}
 
 	renderScreen() {
+		const productsLoaded = ! isEmpty( this.props.productsList );
+
 		return (
 			<div>
+				{ ! productsLoaded && <QueryProductsList /> }
```

Comment:
Is this the correct pattern? Or should we always re-fetch? Seems like the latter is the more common pattern, though I might be wrong. We don't change the products list that often, but if we add a new TLD, we probably want to have it fetched here :)